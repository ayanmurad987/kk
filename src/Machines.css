.listHead{
    display: flex;
    width: 100%;
    justify-content: space-around;
}

.listHead > label {
    font-weight: bold;
}

.listBody {
    display: flex;
    width: 100%;
    justify-content: space-around;
    border: 1px solid;
    padding: 5px;
}

.listBody > label {
    display: flex;
    flex: 1;
    align-items: center;
}