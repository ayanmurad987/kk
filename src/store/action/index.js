import _machines from "./machines";

export const Machines = _machines;
