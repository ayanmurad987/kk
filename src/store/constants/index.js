const Machine = {
	GET_MACHINE_REQUEST: "GET_MACHINE_REQUEST",
	GET_MACHINE_SUCCESS: "GET_MACHINE_SUCCESS",
	GET_MACHINE_FAILED: "GET_MACHINE_FAILED",
	GET_SINGLE_MACHINE_REQUEST: "GET_SINGLE_MACHINE_REQUEST",
	GET_SINGLE_MACHINE_SUCCESS: "GET_SINGLE_MACHINE_SUCCESS",
	GET_SINGLE_MACHINE_FAILED: "GET_SINGLE_MACHINE_FAILED",
	UPDATE_SINGLE_MACHINE_REQUEST: "UPDATE_SINGLE_MACHINE_REQUEST",
	UPDATE_SINGLE_MACHINE_SUCCESS: "UPDATE_SINGLE_MACHINE_SUCCESS",
	UPDATE_SINGLE_MACHINE_FAILED: "UPDATE_SINGLE_MACHINE_FAILED",
};

export default { Machine };
