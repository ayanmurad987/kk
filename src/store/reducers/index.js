import { combineReducers } from 'redux'
import machine from './machine'

export const rootReducer = combineReducers({
    machine
})