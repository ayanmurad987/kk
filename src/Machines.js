import React, { useEffect, useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux'
import { Machines as machinesAction } from './store/action'
import Health from './components/Health'
import { useHistory } from 'react-router-dom'
import './Machines.css'

export default function Machines() {
	const machines = useSelector(state => state.machine.data)
	const dispatch = useDispatch()
	const history = useHistory()

	const getMachines = useCallback(
		() => dispatch(machinesAction.getAll()),
		[dispatch]
	)

	useEffect(() => {
		getMachines()
	}, [])

	const viewDetails = (value) => {
		history.push(`/machines/${value.id}`)
	}

	const renderList = () => {
		return machines.length && machines.map((value, index) =>
			<div key={index} className='listBody' onClick={()=>viewDetails(value)}>
				<label >{value.name}</label>
				<label>{value.ip_address}</label>
				<label><Health value={value.health} /></label>
			</div>
		)
	}

	return (
		<div>
			<div className='listHead'>
				<label>Name</label>
				<label>IP Address</label>
				<label>Health</label>
			</div>
			{renderList()}
		</div>
	);
}
