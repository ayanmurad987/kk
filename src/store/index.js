import { createStore , compose, applyMiddleware} from 'redux'
import logger from 'redux-logger'
import thunk from 'redux-thunk'
import { rootReducer } from './reducers'

const enhancer = compose(applyMiddleware(thunk, logger))
const store = createStore(rootReducer, enhancer)
export default store