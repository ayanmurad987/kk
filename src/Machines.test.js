import React from 'react'
import { shallow, mount } from 'enzyme'
import Machines from './Machines'
import { Provider } from 'react-redux'
import store from './store'

it('renders Machines component without crashing.', () => {
  const wrapper = shallow(
    <Provider store={store}>
      <Machines />
    </Provider>)
  expect(wrapper).toMatchSnapshot()
})
