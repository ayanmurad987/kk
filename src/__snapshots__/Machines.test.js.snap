// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Machines component without crashing. 1`] = `ShallowWrapper {}`;
